### DISCUSSION TITLE
nextjs api route + highlightjs: code highlight not working

### DISCUSSION THREAD

**Discussion by 0xbe1** [vercel/next.js#36407] [Help] [answered]
hi, my app https://startblock-git-highlight-0xbe1.vercel.app wants to offer an experience that: when a user click a "view abi" button, it opens a new tab and visit https://startblock-git-highlight-0xbe1.vercel.app/api/abi?network=ethereum&address=0x25ed58c027921E14D86380eA2646E3a1B5C55A8b, where an HTML is returned and renders the code snippet (in this case, a JSON) using [highlight.js](https://highlightjs.org/) - a code highlighting library.

However, the CSS in the HTML response is all empty, and obviously, there's no code highlighting happening.

```html
<span class="hljs-punctuation">...
```

I wonder if this is an issue with nextjs API route + highlights? How to resolve this? Thank you!

**Comment by icyJoseph** [comment, marked as answer]
API routes are not pages, so the contents of `_app`, and `_document` are not shared, let alone React itself. 

This is good because if I wanted to have other types of content server from `api/routes` for example, robots.txt, then I'd serve true text, rather than text wrapped in the `_document` and `_app` rendered html. 

There's two things you could try here:

- use a `page` with `getServerSideProps`, decode the address query and pass it as props (HTML as a string) to the page, which in turn dangerously set it as inner html for a node (usually a div)
- serve or attach link/script tags on the served HTML pointing to highlight.js [hosted versions](https://highlightjs.org/download/), and don't forget to include a script to do `hljs.highlightAll()`
